### PR TITLE
Conform to naming guidelines and write better code

### PR DESCRIPTION
**Pull request by jdahm**
## Technical Description

No change in code behavior. Simply writing C++ code and conforming to naming rules.

### CODE REVIEW

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

File: dawn/src/dawn/Compiler/DawnCompiler.cpp [hunk 1]
```diff
@@ -87,6 +87,12 @@ struct ComputeEditDistance<std::string> {
                                 : "";
   }
 };
+
+std::string filenameSansExtension(const std::string& fullName, const std::string& extension) {
+  const std::size_t pos = fullName.find_last_of(extension);
+  return pos < std::string::npos ? fullName.substr(0, pos - 3) : fullName;
```
Comment (twicki) [inline dawn/src/dawn/Compiler/DawnCompiler.cpp]: hard coding the 3 here seems risky

**Comment by jdahm** [comment]
launch jenkins

**Pull request merged**